pwiz_tools/Skyline/TestFunctional/PasteMoleculesTest.cs

```diff
@@ -1587,6 +1591,77 @@ private void TestUnsortedMzPrecursors()
 
         }
 
+        private void TestAutoManage()
+        {
+            const string text =
+                "Precursor Name,Precursor Formula,Precursor Adduct,Precursor charge,Explicit Retention Time,Collisional Cross Section (Sq A),Product m/z,product charge,explicit ion mobility High energy Offset,Explicit Collision Energy\n" +
+                "Sulfamethizole,C9H10N4O2S2,[M+H],1,1.85,157.7,,,,1\n" +
+                "Sulfamethizole,C9H10N4O2S2,[M+H],1,1.85,157.7,156.0112,1,0.5,1\n" +
+                "Sulfamethizole,C9H10N4O2S2,[M+H],1,1.85,157.7,92.0498,1,0.51,1\n";
+
+            var docOrig = NewDocument();
+            SetClipboardText(text);
+            // Paste directly into targets area
+            RunUI(() => SkylineWindow.Paste());
+            var pastedDoc = WaitForDocumentChange(docOrig);
+            AssertEx.IsDocumentState(SkylineWindow.Document, null, 1, 1, 1, 3);
+
+            // Because we created nodes with auto manage off, changing these settings should not change the nodes (1 precursor, two fragments)
+            RunUI(() => SkylineWindow.ModifyDocument("Change isotope peaks count and ion types",
+                doc => doc.ChangeSettings(doc.Settings.ChangeTransitionFullScan(fs =>
+                        fs.ChangePrecursorIsotopes(FullScanPrecursorIsotopes.Count, 3, IsotopeEnrichmentsList.GetDefault())).
+                    ChangeTransitionFilter(f => f.ChangeSmallMoleculeIonTypes(new[] { IonType.custom, IonType.precursor })))));
+            pastedDoc = WaitForDocumentChange(pastedDoc);
+            AssertEx.IsDocumentState(SkylineWindow.Document, null, 1, 1, 1, 3);
+
+            var managed = false;
+            for (var loop = 0; loop < 2; loop++)

```

Comment:
You could get rid of the variable "loop" if you did:
`for (var managed in new[] { false, true })`

Reply:
Yes, tidier, thanks.